tic_tac_toe/src/board.py

```diff
@@ -0,0 +1,185 @@
+
+        y : integer or float
+        The first of the two 2-dim coordinates to be operated on
+
+        player : string
+        "X" or "O", which player
+        Returns
+        ---------
+        : boolean
+        Whether or not it is a valid move
+        """
+        return self.values[self.convert(x, y)] == None
+
+    def _check_column(self, x_or_o):
+        """
+        checks if a specific player has won with a column win
+        parameters
+        ----------
+        board : class
+        the board to operate on
+
+        x_or_o : string
+        which player we are checking for
+
+        returns
+        ----------
+        winner : boolean
+        whether or not that player won with a column win
+        """
+        board_width = self.dim
+        winner = None
+
+        for i in range(0, board_width):
+            values = set()
+            for j in range(0, board_width):
+                index = self.convert(i, j)
+                if self.values[index] != None:
+                    values.add(self.values[index])
+            if len(values) == 1:
+                if x_or_o == list(values)[0]:
+                    winner = True
+            else:
+                winner = False
+        return winner
+
+    def _check_row(self, x_or_o):
+        """
+        Checks if a specific player has won with a row win
+        Parameters
+        ----------
+        board : class
+        The board to operate on
+
+        x_or_o : string
+        Which player we are checking for
+
+        Returns
+        ----------
+        winner : boolean
+        Whether or not that player won with a row win
+        """
+        board_width = self.dim
+        winner = None
+        for i in range(0, board_width):
+            values = set()
+            for j in range(0, board_width):
+                index = self.convert(j, i)
+                if self.values[index] != None:
+                    values.add(self.values[index])
+            if len(values) == 1:
+                if x_or_o == list(values)[0]:
+                    winner = True
+            else:
+                winner = False
+        return winner
+
+    def _check_diagonal(self, x_or_o):
+        """
+        Checks if a specific player has won with a diagonal win
+        Parameters
+        ----------
+        board : class
+        The board to operate on
+
+        x_or_o : string
+        Which player we are checking for
+        Returns
+        ----------
+        winner : boolean
+        Whether or not that player won with a diagonal win
+        """
+
+        # Check of someone has won
+        board_width = self.dim
+        winner = None
+        values = set()
+        values2 = set()
+        for i in range(0, board_width):
+            index = self.convert(i, i)
+            index2 = self.convert(i, (-i + 2))
+            values.add(self.values[index])
+            values2.add(self.values[index2])
+        if len(values) == 1:
```

Comment:
These two blocks of code are identical with the exception that they're looking at `values` and `values2`.  Can you generalize?  Perhaps with an internal function to `_check_diagonal`?